code/modules/clothing/suits/wiz_robe.dm

```diff
@@ -221,9 +220,7 @@
 
 /obj/item/wizard_armour_charge/afterattack(obj/item/clothing/suit/space/hardsuit/shielded/wizard/W, mob/user)
 	. = ..()
-	if(!istype(W))
-		to_chat(user, "<span class='warning'>The rune can only be used on battlemage armour!</span>")
+	if(SEND_SIGNAL(W, COMPONENT_RECHARGE_SHIELD, user, 8))

```

Comment:
Don't do this. Just listen to COMSIG_ATOM_ATTACKBY in the component should item_rechargable not be empty.